### PR TITLE
finished tilesetmenu

### PR DESCRIPTION
**Pull request by DanielLiebler**

### CODE REVIEW

File: src/app/components/lobby/dialogs/open-game-popup/open-game-popup.component.ts [hunk 1]
```diff
@@ -67,37 +74,40 @@ export class OpenGamePopupComponent implements OnInit {
...
+  tileListAsSet: Map<number, { bt: BoardTile; count: number }> = new Map<number, { bt: BoardTile; count: number }>();
   popoverVisible = false;
 
-  @ViewChild('deckCarousel', {static : true}) deckCarousel: NgbCarousel;
-
+  @ViewChild('deckCarousel', { static: true }) deckCarousel: NgbCarousel;
 
   // Onclick FieldList
   //  multiples as x2 - tag
   //  only list of pictures
```
Comment (Saphs) [inline src/app/components/lobby/dialogs/open-game-popup/open-game-popup.component.ts:84]: please remove useless comments. Or if they are needed please elaborate what they seem to be saying as I do not understand this one.

File: src/app/components/lobby/dialogs/open-game-popup/open-game-popup.component.ts [hunk 2]
```diff
@@ -67,37 +74,40 @@ export class OpenGamePopupComponent implements OnInit {
...
-    });
+    req.subscribe(
+      (res: APIResponse<TileSet[]>) => {
+        const ts = res.payload;
+        ts.forEach((val: TileSet) => {
+          if (val.thumbnailPath === undefined || val.thumbnailPath === null) {
+            val.thumbnailPath = OpenGamePopupComponent.defaultThumbPath;
+          }
+        });
+        this.deckList = ts;
```
Comment (Saphs) [inline src/app/components/lobby/dialogs/open-game-popup/open-game-popup.component.ts:104]: This naming is confusing, could we change this to be atleast consistent?

File: src/app/components/lobby/dialogs/open-game-popup/open-game-popup.component.ts [hunk 3]
```diff
@@ -116,37 +126,39 @@ export class OpenGamePopupComponent implements OnInit {
...
     this.dialogRef.close();
   }
 
   public createGame() {
     if (this.roomName !== undefined) {
       this.dialogRef.close({
         roomName: this.roomName,
-        skinName: (this.skinName !== undefined && this.skinName.length > 0) ? this.skinName : 'default',
-        randomizeTiles: this.randomizeTiles || false
+        tileSetId: this.deckList[this.selectedDeck].id || 1,
```
Comment (Saphs) [inline src/app/components/lobby/dialogs/open-game-popup/open-game-popup.component.ts:151]: dont really understand why this resolution is nessecary? I thought this.selectedDeck was already the Id, why do we need to resolve it again?

**Review by Saphs — COMMENTED** [review]
I feel it would drastically increases the review experience if you could seperate formatting and actual code changes in commits.

File: src/app/components/lobby/dialogs/open-game-popup/open-game-popup.component.ts [hunk 4]
```diff
@@ -161,7 +157,7 @@ export class OpenGamePopupComponent {
 
   slid(evt: NgbSingleSlideEvent, slide: number): void {
     if (evt.isShown) {
-      this.selectedDeck = slide;
+      this.selectedTileSetId = slide;
       const elmnt = document.getElementById('deck-list-entry-' + slide);
       console.log('elmnt is', elmnt);
       // elmnt.scrollIntoView();
```
Comment (Saphs) [inline src/app/components/lobby/dialogs/open-game-popup/open-game-popup.component.ts:163]: meh

**Comment by Saphs** [comment]
Please use rebases to fix your commit structure.

**Comment by DanielLiebler** [comment]
> Please use rebases to fix your commit structure.

When rebasing, a force-push would be required.
Force-pushes are bad behavior. Therefore, I wont do that

**Review by Saphs — APPROVED** [review]

**Pull request merged**